src/main/java/by/custom/utilcalculator/domain/UserProgressStorage.java

```diff
@@ -1,27 +1,52 @@
 package by.custom.utilcalculator.domain;
 
+import org.telegram.telegrambots.meta.api.objects.Message;
+
+import java.io.*;
 import java.util.HashMap;
 
 public class UserProgressStorage {
-    private final HashMap<String, UserProgress> users;
+
+    private static class UserProgressStorageHolder {
+        private static final UserProgressStorage USER_PROGRESS_STORAGE = new UserProgressStorage();
+    }
+
+    private HashMap<String, UserProgress> users;
```

Comment:
I recommend removing `Map` and storing data only in file(s). I remember that you liked idea of storing every user in separate file. Maybe you could implement it?

Reply:
Realized. It was easyer than I thought. I am not sure about the code, but at least it seems to work. I am happy. Thanks for yout support!